src/code/code_800EC960.c

```diff
@@ -1203,916 +1233,1080 @@ u8 D_8016B9F4;
-    sOcarinaSongs[OCARINA_SONG_MEMORY_GAME][sOcaMinigameAppendPos].tone = 0;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos].pitch = randomPitch;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos].length = 45;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos].volume = 0x50;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos].vibrato = 0;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos].bend = 0;
 
-    sOcaMinigameAppendPos++;
+    sOcaMemoryGameAppendPos++;
 
-    sOcarinaSongs[OCARINA_SONG_MEMORY_GAME][sOcaMinigameAppendPos].noteIdx = 0xFF;
-    sOcarinaSongs[OCARINA_SONG_MEMORY_GAME][sOcaMinigameAppendPos].unk_02 = 0;
-    sOcarinaSongs[OCARINA_SONG_MEMORY_GAME][sOcaMinigameAppendPos + 1].noteIdx = 0xFF;
-    sOcarinaSongs[OCARINA_SONG_MEMORY_GAME][sOcaMinigameAppendPos + 1].unk_02 = 0;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos].pitch = OCARINA_PITCH_NONE;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos].length = 0;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos + 1].pitch = OCARINA_PITCH_NONE;
+    sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME][sOcaMemoryGameAppendPos + 1].length = 0;
     if (1) {}
-    return 0;
+    return false;
 }
 
-// input update?
-void func_800EE6F4(void) {
-    D_8016BA04 = gAudioContext.totalTaskCnt;
-    if (D_80130F10 != 0) {
-        if (sOcarinaInpEnabled == 1) {
-            Audio_GetOcaInput();
+void AudioOcarina_Update(void) {
+    sOcarinaUpdateTaskStart = gAudioContext.totalTaskCount;
+    if (sOcarinaInstrumentId != OCARINA_INSTRUMENT_OFF) {
+        if (sIsOcarinaInputEnabled == true) {
+            AudioOcarina_ReadControllerInput();
         }
-        if ((sPlaybackState == 0) && (sOcarinaInpEnabled == 1)) {
-            func_800ED458(0);
+
+        if ((sPlaybackState == 0) && (sIsOcarinaInputEnabled == true)) {
+            AudioOcarina_PlayControllerInput(false);
         }
-        if (D_80130F3C != 0) {
-            if (D_80130F3C & 0x4000) {
-                func_800ED200();
+
+        if (sOcarinaFlags != 0) {
+            if (sOcarinaFlags & 0x4000) {
+                AudioOcarina_CheckSongsWithoutMusicStaff();
             } else {
-                func_800ECDF8();
+                AudioOcarina_CheckSongsWithMusicStaff();
             }
         }
 
-        Audio_OcaPlayback();
-        D_80130F68 = D_8016BA04;
+        AudioOcarina_PlaybackSong();
+        sOcarinaPlaybackTaskStart = sOcarinaUpdateTaskStart;
 
         if (sPlaybackState == 0) {
-            func_800EE404();
+            AudioOcarina_RecordSong();
         }
 
-        if ((D_80130F3C != 0) && (sPrevOcarinaNoteVal != sCurOcarinaBtnVal)) {
-            D_80131880 = 1;
+        if ((sOcarinaFlags != 0) && (sPrevOcarinaPitch != sCurOcarinaPitch)) {
+            sIsOcarinaNoteChanged = true; // Adds a 1 frame delay to ocarina input
```

Comment:
I think it drops inputs for one frame, not delays
I'm also not enthusiastic about this being a boolean, I see this value more as a timer